File: 2022/4/day4.py
import re

with open("input/day4_input.txt") as f:
    input = f.read()

input = re.split(",|-|\n", input)
input = [int(x) for x in input]

count = 0
for i in range(0, len(input), 4):
    if input[i] <= input[i + 2] and input[i + 1] >= input[i + 3]:
        count += 1
    elif input[i] >= input[i + 2] and input[i + 1] <= input[i + 3]:
        count += 1

print(count)

count = 0
for i in range(0, len(input), 4):
    if input[i] <= input[i + 2] and input[i + 1] >= input[i + 2]:
        count += 1
    elif input[i + 2] <= input[i] and input[i + 3] >= input[i]:
        count += 1

print(count)

File: 2022/4/day4.c
#include <stdio.h>

int main()
{
    FILE *file = fopen("input/day4_input.txt", "r");
    int x[4];
    int count1 = 0, count2 = 0;

    while (!feof(file))
    {
        fscanf(file, "%d-%d,%d-%d\n", &x[0], &x[1], &x[2], &x[3]);

        if ((x[0] <= x[2] && x[1] >= x[3]) || (x[0] >= x[2] && x[1] <= x[3]))
            count1++;
        if ((x[0] <= x[2] && x[1] >= x[2]) || (x[2] <= x[0] && x[3] >= x[0]))
            count2++;
    }

    printf("Sol1: %d\n", count1);
    printf("Sol2: %d\n", count2);

    return 0;
}